### ISSUE TITLE
Need help using it with Bluestacks Pie 64-bit

### ISSUE THREAD

**Issue by jens4626** [pal1000/mesa-dist-win#107]
I have been using Mesa3D running Bluestacks Nougat 32-bit, and it worked fine with OpenGL. Sadly The game I used it for recently updated, so now it requires Bluestacks to be Pie 64-bit, and I can't get it to work with that. The app opens and runs fine with a GPU, but the game is completely black, and doesn't load using your mesa dist win. Perhaps you can test if I did something wrong, or it doesn't support it - if it doesn't support it, I hope you will look into it!

Here is a few informations you perhaps needs:

Bluestacks 5: https://www.bluestacks.com/bluestacks-5.html
Bluestacks instance needs to be Android version Pie 64-bit (Beta) 
Graphics render in Bluestacks needs to be OpenGL

Shortcut to opendll location: C:\Program Files\BlueStacks_nxt

You can also simply install the game in Bluestacks from Play Store "Old School Runescape"

I really hope you can help me out!

**Comment by pal1000** [comment]
> You can also simply install the game in Bluestacks from Play Store "Old School Runescape"

Is it [this game](https://play.google.com/store/apps/details?id=com.jagex.oldscape.android&gl=US)?

What's the GPU in use as reported by Bluestacks in Settings - Graphics?

For me it was llvmpipe and I was able to reproduce the black screen with Mesa 22.1.3 MSVC build.

**Comment by jens4626** [comment]
> > You can also simply install the game in Bluestacks from Play Store "Old School Runescape"
> 
> Is it [this game](https://play.google.com/store/apps/details?id=com.jagex.oldscape.android&gl=US)?
> 
> What's the GPU in use as reported by Bluestacks in Settings - Graphics?
> 
> For me it was and I was able to reproduce the black screen with Mesa 22.1.3 MSVC build.

Yea that’s the game. It is also llvmpipe for me. Is there something you can do to solve this issue with black screen? It must have something to do with your project / mesa since the game can load without it (if you have a gpu - but I want to use it in a pc without gpu)

**Comment by pal1000** [comment]
I tried DirectX renderer with my GPU and got the same black screen. Last hope is zink driver over lavapipe or swiftshader.  If neither work we'll have to report it [upstream](https://gitlab.freedesktop.org/mesa/mesa/-/issues) but don't expect a fix anytime soon.

**Comment by jens4626** [comment]
> I tried DirectX renderer with my GPU and got the same black screen. Last hope is zink driver over lavapipe or swiftshader. If neither work we'll have to report it [upstream](https://gitlab.freedesktop.org/mesa/mesa/-/issues) but don't expect a fix anytime soon.

The game runs on OpenGL, so it needs to be that - DirectX won’t work for that reason, no matter what. The zink driver over lava pipe or swiftshader is that something you can test?

**Comment by pal1000** [comment]
> The zink driver over lava pipe or swiftshader is that something you can test?

Tried and both led to crash. I'll also try GLonD3D12 over WARP. I don't have much hope though.

**Comment by pal1000** [comment]
GLonD3D12 over WARP = black screen in game like llvmpipe. So this is disappointing as i tried all Mesa3D OpenGL drivers. As an irony the latest graphics driver (9.17.10.4459) for this end of life  iGPU (Intel HD Graphics 3000) of this old PC of mine was able to display the in-game login screen correctly despite the rendering of Android UI being a disaster.

**Comment by Shadiezz** [comment]
Yeah I also gave it a shot with the Mesa3D OpenGL drivers and are also just getting black screens.

**Comment by jens4626** [comment]
> GLonD3D12 over WARP = black screen in game like llvmpipe. So this is disappointing as i tried all Mesa3D OpenGL drivers. As an irony the latest graphics driver (9.17.10.4459) for this end of life iGPU (Intel HD Graphics 3000) of this old PC of mine was able to display the in-game login screen correctly despite the rendering of Android UI being a disaster.

I’m not really sure what to do from here, but I’m hoping you will forward this to those who might be able to solve it? Would be really awesome

**Comment by pal1000** [comment]
> I’m not really sure what to do from here, but I’m hoping you will forward this to those who might be able to solve it?

Done.

**Comment by jens4626** [comment]
Any updates? :)

**Comment by pal1000** [comment]
Unfortunately no, but I'll re-test to see if anything changed soon.

**Comment by jens4626** [comment]
FYI I just tried to reinstall bluestacks but it appears to crash before startup using your software. I tried version 22.1.3 (which worked like 2 months ago) - 22.1.7. 

I tried using this, and it can load bluestacks without crash, but it gives the blackscreen in game. 
https://fdossena.com/?p=mesa/index.frag

Fdossena doesn't crash bluestacks, so perhaps this will help you to figure out why bluestacks can't even startup anymore with yours.

**Comment by jezzyjax** [comment]
Any update

**Comment by jens4626** [comment]
> Unfortunately no, but I'll re-test to see if anything changed soon.

Did you have the time to test this yet?
